src/index.js

```diff
@@ -16,21 +17,36 @@ const {
   APPROVE_ONLY,
   API_URL,
   TARGET,
+  PR_NUMBER,
 } = getInputs()
 
 const GITHUB_APP_URL = 'https://github.com/apps/dependabot-merge-action'
 
 async function run() {
   try {
-    const { pull_request: pr } = github.context.payload
+    const { pull_request, workflow } = github.context.payload
 
-    if (!pr) {
+    const isSupportedContext = pull_request || workflow
+
+    if (!isSupportedContext) {
       return logError(
-        'This action must be used in the context of a Pull Request'
+        'This action must be used in the context of a Pull Request or a Workflow Dispatch event'
       )
     }
 
-    const pullRequestNumber = pr.number
+    let pr = pull_request
+
+    const pullRequestNumber = PR_NUMBER || pr.number
```

Comment:
this may throw an exception if the pr-number input is not provided and this is not run in the context of a pull request. make it more robust

Reply:
I've added a check to log an error if a PR number isn't present.

Looks like `core.getInput` will return as an empty string for an optional input, so this should be okay?

https://github.com/actions/toolkit/blob/2f164000dcd42fb08287824a3bc3030dbed33687/packages/core/src/core.ts#L69-L77

Reply:
this is not the problem. if PR_NUMER is falsy and github.context.payload.pull_request is falsy, it will throw a JavaScript error

Reply:
I've included a check after this statement since a pull request is required to do any of the later work. Is that not enough? I can add a check before I do the variable assignment to `pullRequestNumber` if that is better?

```javascript
const pullRequestNumber = PR_NUMBER || pr.number

if (!pullRequestNumber) {
  return logError(
    'No pull request number has been found. Please make sure a pull request number has been provided'
  )
}
```

Reply:
I suppose I can introduce a variable `hasPullRequest`? Then I can check the value of that, and error out if nothing is present or defined.

```javascript
const hasPullRequest = pull_request || PR_NUMBER
```

Reply:
The action needs to be robust enough that wrong user input doesn't lead to unexpected errors

Reply:
@simoneb okay check it out now, I think there are enough checks and conditionals to prevent an issue based on user input or context.